compiler/env/OMRAotAdapter.cpp

```diff
@@ -0,0 +1,142 @@
+ * Copyright (c) 2000, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "env/AotAdapter.hpp"
+#include "env/CompilerEnv.hpp"
+#include "runtime/CodeCacheManager.hpp"
+// #include "env/SharedCache.hpp"
+#include "runtime/CodeCache.hpp"
+#include <iostream>
+#include "runtime/RelocationRuntime.hpp"
+
+TR::AotAdapter *
+OMR::AotAdapter::self()
+    {
+    return static_cast<TR::AotAdapter *>(this);
+    }
+ void OMR::AotAdapter::initializeAOTClasses(TR::RawAllocator* rawAllocator,TR::CodeCacheManager* cc){
+//   _sharedCache = new (PERSISTENT_NEW) TR::SharedCache("wasm_shared_cache", "/tmp");
+  _reloRuntime = new (PERSISTENT_NEW) TR::SharedCacheRelocationRuntime (NULL);
+  _codeCacheManager = cc;
+ }
+ 
+void OMR::AotAdapter::storeExternalSymbol(const char *symbolName, void* symbolAddress){
+    _reloRuntime->registerLoadedSymbol(symbolName,symbolAddress);
+}
+
+
+void OMR::AotAdapter::storeAOTMethodAndDataInTheCache(const char* methodName){
+    TR::AOTMethodHeader* hdr =_methodNameToHeaderMap[methodName];
+    // _sharedCache->storeEntry(methodName,hdr->serialize(),hdr->sizeOfSerializedVersion());
+}
+
+TR::AOTMethodHeader* OMR::AotAdapter::loadAOTMethodAndDataFromTheCache(const char* methodName)
+{
+    TR::AOTMethodHeader* methodHeader = NULL;
+    void* cacheEntry;
+    // void* cacheEntry = reinterpret_cast<uint8_t*> (_sharedCache->loadEntry(methodName));
+    if (cacheEntry!=NULL){
+        methodHeader = new TR::AOTMethodHeader((uint8_t*)cacheEntry);
+        void* codeStart = methodHeader->compiledCodeStart;
+        _reloRuntime->registerLoadedSymbol(methodName, codeStart);
+        registerAOTMethodHeader(methodName,methodHeader);
+    }
+
+    return methodHeader;
+}
+void OMR::AotAdapter::registerAOTMethodHeader(std::string methodName,TR::AOTMethodHeader* header){
+    _methodNameToHeaderMap[methodName] = header;
+}
```

Comment:
What is the purpose of `registerLoadedSymbol` and `registerAOTMethodHeader`? I think the distinction needs to be documented.

Reply:
Some comments were introduced to codebase and will be shipped soon, thank you for pointing that out!  Once before accepted, I will go through the code, documenting it better.